agent/grpc-external/options_test.go

```diff
@@ -0,0 +1,35 @@
+package external
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/metadata"
+)
+
+func TestQueryOptionsFromContextRoundTrip(t *testing.T) {
+
+	expected := structs.QueryOptions{
+		Token: "123",
```

Comment:
Could you test all the fields on `QueryOptions`. I think `QueryOptionsFromContext` might result in errors when non-string fields are set.

Reply:
Great catch, thanks! I updated the decode logic.